bgpd/bgp_attr.c

```diff
@@ -3635,10 +3636,21 @@ size_t bgp_packet_mpattr_start(struct stream *s, struct peer *peer, afi_t afi,
 			stream_put_ipv4(s, attr->nexthop.s_addr);
 			break;
 		case SAFI_MPLS_VPN:
-			stream_putc(s, 12);
-			stream_putl(s, 0); /* RD = 0, per RFC */
-			stream_putl(s, 0);
-			stream_put(s, &attr->mp_nexthop_global_in, 4);
+			plk = peer_lookup_by_host(NULL, peer->host);
+			if (attr->srv6_l3vpn && plk && plk->update_source &&
+			    plk->update_source->sin6.sin6_family == AF_INET6) {
```

Comment:
Is it possible to have AF_INET here for sockaddr_in6?

Reply:
This is common part of struct. Check if source is IPV6 (id 10).